nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/storage/AzureDataLakeStorageFileResourceService.java

```diff
@@ -0,0 +1,139 @@
+@SeeAlso({FetchAzureDataLakeStorage.class})
+@CapabilityDescription("Provides an Azure Data Lake Storage (ADLS) file resource for other components.")
+@UseCase(
+        description = "Fetch the specified file from Azure Data Lake Storage." +
+                " The service provides higher performance compared to fetch processors when the data should be moved between different storages without any transformation.",
+        configuration = """
+                "Filesystem Name" = "${azure.filesystem}"
+                "Directory Name" = "${azure.directory}"
+                "File Name" = "${azure.filename}"
+
+                The "ADLS Credentials" property should specify an instance of the ADLSCredentialsService in order to provide credentials for accessing the filesystem.
+                """
+)
+public class AzureDataLakeStorageFileResourceService extends AbstractControllerService implements FileResourceService {
+
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            ADLS_CREDENTIALS_SERVICE,
+            FILESYSTEM,
+            DIRECTORY,
+            FILE
+    );
+
+    private volatile DataLakeServiceClientFactory clientFactory;
+    private volatile ConfigurationContext context;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        this.clientFactory = new DataLakeServiceClientFactory(getLogger(), getProxyOptions(context));
+        this.context = context;
+    }
+
+    @OnDisabled
+    public void onDisabled() {
+        this.clientFactory = null;
+        this.context = null;
+    }
+
+    @Override
+    public FileResource getFileResource(Map<String, String> attributes) {
+        final DataLakeServiceClient client = getStorageClient(attributes);
+        try {
+            return fetchBlob(client, attributes);
+        } catch (final DataLakeStorageException | IOException e) {
+            throw new ProcessException("Failed to fetch file from ADLS Storage", e);
+        }
+    }
+
+    protected DataLakeServiceClient getStorageClient(Map<String, String> attributes) {
+        final ADLSCredentialsService credentialsService = context.getProperty(ADLS_CREDENTIALS_SERVICE)
+                .asControllerService(ADLSCredentialsService.class);
+        return clientFactory.getStorageClient(credentialsService.getCredentialsDetails(attributes));
+    }
+
+    /**
+     * Fetching file from the provided filesystem and directory in ADLS.
+     *
+     * @param storageClient azure data lake service client
+     * @param attributes configuration attributes
+     * @return fetched file as FileResource
+     * @throws IOException exception caused by missing parameters or blob not found
+     */
+    private FileResource fetchBlob(final DataLakeServiceClient storageClient, final Map<String, String> attributes) throws IOException {
```

Comment:
The method should rather be called `fetchFile()` as it is referred to in its javadoc.